File: app/src/main/java/com/example/nutriTrack/PostsListViewModel.java
package com.example.nutriTrack;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;


import com.example.nutriTrack.Model.Post;

import java.util.List;

public class PostsListViewModel extends ViewModel {

    public enum ListMode {
        AllPosts,
        MyPosts
    }

    LiveData<List<Post>> data;
    ListMode mode;

    public PostsListViewModel(ListMode listMode) {
        mode = listMode;

//        if (listMode == ListMode.AllPosts) {
//            data = Model.instance().getAllMaslulim();
//        } else {
//            data = Model.instance().getMyMaslulim();  // TODO: add model
//        }
    }

    public LiveData<List<Post>> getData() {
        return data;
    }

    public ListMode getMode() {
        return this.mode;
    }
}

File: app/src/main/java/com/example/nutriTrack/PostsListViewModel.kt
package com.example.nutriTrack.ViewModel

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.example.nutriTrack.Model.Model
import com.example.nutriTrack.Model.Post

class PostsListViewModel : ViewModel() {
    var posts: LiveData<List<Post>> = Model.shared.posts

//    fun getPosts(userId: String?): LiveData<List<Post>> {
//        return if (userId == null) {
//           Model.getAllPosts()
//        } else {
//            Model.getPostsByUserId(userId)
//        }
//    }

    fun refreshAllPost() {
        Model.shared.refreshAllPosts()
    }
}